### ISSUE TITLE
Run as a release on Ubuntu 16.04 / libstdc++.so.6 reported missing

### ISSUE THREAD

**Issue by raju-bitter** [flutter/flutter#6207]
After creating a helloworld project, tried to run the app on Android as a release on Ubuntu. Launching in debug mode worked fine, but when I tried to run as a release, I got an error that 'libstdc++.so.6' is missing.

**1) Launching with missing library**
raju@titan:~/coding/flutter/helloworld$ flutter run --release --full-restart
Building APK in release mode (android-arm-release)...  
Dart snapshot generator failed with exit code 127
/home/raju/dev/flutter/bin/cache/artifacts/engine/android-arm-release/linux-x64/gen_snapshot: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
Failed to build AOT snapshot

**2) Installing the missing package**
raju@titan:~/coding/flutter/helloworld$ sudo apt-get install lib32stdc++6
[sudo] password for raju: 
Reading package lists... Done
Building dependency tree  
Reading state information... Done
The following NEW packages will be installed:
  lib32stdc++6
0 upgraded, 1 newly installed, 0 to remove and 5 not upgraded.
Need to get 0 B/404 kB of archives.
After this operation, 1,901 kB of additional disk space will be used.
Selecting previously unselected package lib32stdc++6.
(Reading database ... 253193 files and directories currently installed.)
Preparing to unpack .../lib32stdc++6_5.4.0-6ubuntu1~16.04.2_amd64.deb ...
Unpacking lib32stdc++6 (5.4.0-6ubuntu1~16.04.2) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Setting up lib32stdc++6 (5.4.0-6ubuntu1~16.04.2) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...

**3) Running as release again**
raju@titan:~/coding/flutter/helloworld$ flutter run --release --full-restart
Building APK in release mode (android-arm-release)...  
Warning: signing the APK using the debug keystore.
Built build/app.apk (8.1MB).
Running lib/main.dart on SM N920S...
--------- beginning of main
Application running.

**Solution**
The _flutter doctor_ tool could check if the library has been installed, or it could be added to the list of required tools here: https://flutter.io/setup/#system-requirements

**Steps to reproduce: System setup and tool installation performed**
1) Install Ubuntu 16.04 using this image
http://releases.ubuntu.com/16.04/ubuntu-16.04-desktop-amd64.iso

2) Install Flutter following these instructions
https://flutter.io/setup/#get-the-flutter-sdk
sudo sh -c 'curl https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'
sudo sh -c 'curl https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'
sudo apt-get update
sudo apt-get install dart

3) Install Dart SDK for Atom
https://www.dartlang.org/install/linux

3) Install Oracle Java and Android Studio with SDK
a) Java
sudo add-apt-repository ppa:webupd8team/java
sudo apt-get update
sudo apt install oracle-java8-installer
sudo apt install oracle-java8-set-default
b) Android Studio
Downloaded deb from website
Installed Android Nougat SDK

4) Install Atom
Downloaded deb file from atom.io
Installed deb

a) Dart & Flutter for Atom
apm install flutter
apm install dartlang

5) `flutter doctor` output now
raju@ubuntu:~/coding$ flutter doctor
[✓] Flutter (on Linux, channel master)
    • Flutter at /home/raju/dev/flutter
    • Framework revision 9f673ad445 (5 hours ago), 2016-10-04 08:20:24
    • Engine revision 27d012a790
    • Tools Dart version 1.20.0-dev.5.0

[✓] Android toolchain - develop for Android devices (Android SDK 24.0.3)
    • Android SDK at /home/raju/Android/Sdk
    • Platform android-24, build-tools 24.0.3
    • Java(TM) SE Runtime Environment (build 1.8.0_101-b13)

[✓] Atom - a lightweight development environment for Flutter
    • flutter plugin version 0.2.6
    • dartlang plugin version 0.6.40

[✓] Connected devices
    • SM N920S • 05157df5d93b793d • android-arm

6) Create new app
flutter create helloworld
cd helloworld
flutter run

**Comment by eseidelGoogle** [comment]
@danrubel should we update flutter doctor to check if `sky_snapshot` can run without errors?  Not sure what the best resolution here is.  This seems like a linux-specific `flutter` setup issue, which could either be solved by documentation or tooling, unsure.

**Comment by chinmaygarde** [comment]
Fortunately, we just switched to using the Fuchsia buildtools. They provide a libcxx and libcxxabi that we can statically link into `gen_snapshot`. However, this only works for x64 targets right now. `gen_snapshot` is i386 because it needs to generate armv7 code. I had filed https://fuchsia.atlassian.net/browse/TO-61 to track the issue related to making libcxx available to us on i386 as well. Now that I know there are actual issues with users not having the `.so` on their system already, I will raise the priority on the same.

A workaround for this issue could be to ensure something like `apt-get install libstdc++6` has already been done on Linux hosts (am on Mac right now so that is from memory).

**Comment by raju-bitter** [comment]
` sudo apt-get install lib32stdc++6` was the command I had to use on 16.04 64bit.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/6040 was another report of this.

**Comment by eseidelGoogle** [comment]
I think https://github.com/flutter/flutter/issues/4916 is another report of this. :/  This is probably something `flutter doctor` should try to catch, given that we've had 3 reports now.  FYI @danrubel

**Comment by ethael** [comment]
I had the same issue. Debian Sid amd64

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/6040 was another report of this.  So I guess that's 4 reports of this mis-configuration on linux?  (That's shockingly high if true.)

**Comment by eseidelGoogle** [comment]
This would be an easy check to add to `flutter doctor`.  We should just do it.  

Probably belongs in:
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/android/android_workflow.dart
since we don't have a linux-specific setup file at the moment.

**Comment by eseidelGoogle** [comment]
#8059 looks very similar, but not maybe identical.  Clearly there are a bunch of packages `flutter doctor` should check for on linux which are not installed by default (at least not on 64-bit installs).  In #8059 @abarth noted that the Android toolchain requires ability to execute 32-bit binaries for instance.

**Comment by mit-mit** [comment]
Moving to milestone 2; this looks like a very common issue on linux boxes.

**Comment by uncleGena** [comment]
if some body facing this problem
**✗ Downloaded executables cannot execute on host (see https://github.com/flutter/flutter/issues/6207 for more information)**
This is that information -> `sudo apt-get install lib32stdc++6`
helped on ubuntu 14.04

**Comment by perlatus** [comment]
On Arch Linux the fix is to enable multilib and install `gcc-libs-multilib`, which replaces `gcc-libs`.

**Comment by txthinking** [comment]
How to solve this on MacOS
```
[-] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel alpha)
    • Flutter at /Users/tx/flutter
    • Framework revision b156a0f054 (3 weeks ago), 2017-08-11 22:01:40 -0700
    • Engine revision fef7d827d6
    • Tools Dart version 1.25.0-dev.9.0
    ✗ Downloaded executables cannot execute on host (see https://github.com/flutter/flutter/issues/6207 for more information)

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /usr/local/opt/android-sdk
    • Platform android-25, build-tools 25.0.2
    • ANDROID_HOME = /usr/local/opt/android-sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version openjdk version "1.8.0_112-release"

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.3.1

[✓] Android Studio (version 2.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version openjdk version "1.8.0_112-release"

[✓] Connected devices
    • None
```

**Comment by eseidelGoogle** [comment]
I wasn't aware this error was possible on Mac OS X. :)  @cbracken would know.

**Comment by porjo** [comment]
On Fedora, `dnf install libstdc++.i686`

**Comment by cbracken** [comment]
@txthinking can you provide details of the error you're seeing on macOS in a new bug? The specific issue in this case is Linux-specific since it relates to the absence of ELF `.so` shared objects -- macOS uses Mach-O `.dylib`s. 

Took a quick look at `gen_snapshot` on macOS, and at least in today's builds, they link to `/usr/lib/libc++.1.dylib`, which should ship with the base macOS install since Apple switched to Clang, which was OS X 10.7 (IIRC).

**Comment by txthinking** [comment]
@cbracken  Here is the new issue #12582

**Comment by nicks189** [comment]
For me on Arch installing `lib32-libstdc++5` worked.

**Comment by aswathm78** [comment]
# Fix for ubuntu 17.04 version users 

Just want to add a fix that worked for me .
I was using lubuntu 17.04 which i think the support was stopped .
After that I upgraded lubuntu using `sudo do-release-upgrade`
With a single restart I got new version

After that I installed clang . It worked 
and I ran this command 
`sudo apt install lib32stdc++6`

And ran flutter doctor again .

![screenshot from 2018-01-24 12-34-28](https://user-images.githubusercontent.com/13545075/35318808-017c0008-0103-11e8-8784-3b93d66fc8ad.png)

works fine now

**Comment by showcheap** [comment]
OpenSUSE user can use this command to fix

```
sudo zypper in libstdc++6-32bit
```

**Comment by Sprunjer** [comment]
installing lib32stdc++6 fixed for Debian 9 as well.

**Comment by eseidelGoogle** [comment]
If someone is interested in changing the `flutter doctor` command to print a more specific message (or installation instructions) I expect such PRs would be welcome: https://github.com/flutter/flutter/blob/26102c9120ebc01b5c5e48c42a01da9f3e849daf/packages/flutter_tools/lib/src/doctor.dart#L249 is the code in question.

**Comment by tvpmkc** [comment]
Getting this now in Windows when running "flutter doctor -v"
[!] Flutter (Channel master, v0.1.7, on Microsoft Windows [Version 10.0.16299.248], locale en-US)
    • Flutter version 0.1.7 at G:\Projects\flutter
    • Framework revision 3b84503403 (21 hours ago), 2018-02-26 20:11:11 -0800
    • Engine revision ead227f118
    • Dart version 2.0.0-dev.28.0.flutter-0b4f01f759
    X Downloaded executables cannot execute on host (see https://github.com/flutter/flutter/issues/6207 for more information)

**Comment by jaworanski** [comment]
@tvpmkc I ran into this issue and it was my antivirus blocking \flutter\bin\cache\artifacts\engine\android-arm-profile\windows-x64\gen_snapshot.exe. Delete the cache folder in \flutter\bin and then run flutter doctor. Check for the exe in the above location, if it’s not there then check your antivirus settings. Hope this helps!

**Comment by cbracken** [comment]
In #15004 I've added instructions for Debian/Ubuntu/Mint/Fedora/Arch to the doctor error message.

**Comment by tvpmkc** [comment]
@jaworanski Spot on. That was the issue. Anti virus was quarantining the exe automatically. It's working now. Thank you!

**Comment by cbracken** [comment]
Great to hear -- glad this got you unblocked!

**Comment by jlelse** [comment]
Just for other **Solus** user, this fixed it for me:

`sudo eopkg it libgcc-32bit libstdc++-32bit`

**Comment by 0x68** [comment]
Solution for Fedora 26/27

`sudo dnf install libstdc++.i686`

**Comment by mohasu1992** [comment]
~ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel beta, v0.1.5, on Linux, locale en_US.utf8)
    ✗ Downloaded executables cannot execute on host (see https://github.com/flutter/flutter/issues/6207 for more information)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] Android Studio (version 3.0)
[✓] Connected devices (1 available)

! Doctor found issues in 1 category.
for SOLUS 3.99

**Comment by Amarog** [comment]
Getting error in Debian (Kali linux) when running flutter doctor,

$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel beta, v0.1.5, on Linux, locale en_IN)
    **✗ Downloaded executables cannot execute on host (see https://github.com/flutter/flutter/issues/6207 for more information)**
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] Android Studio (version 3.0)
[!] IntelliJ IDEA Ultimate Edition (version 2017.3)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[!] Connected devices
    ! No devices available

@Sprunjer said _Installing lib32stdc++6 fixed it for Debian 9 as well._
but when running sudo apt get install lib32stdc++6, it is trying to remove multiple packages.. i doubt it might create problems to other s/w in system.

$ sudo apt-get install lib32stdc++6
[sudo] password for jwx: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  hashcat-data libasan3 libboost-atomic1.62.0 libboost-chrono1.62.0
  libboost-program-options1.62.0 libboost-serialization1.62.0
  libboost-test1.62.0 libboost-timer1.62.0 libc-dev-bin libcgal12
  libclang-common-3.8-dev libclang1-3.8 libevent-2.0-5 libffi-dev
  libfreerdp-cache1.1 libfreerdp-client1.1 libfreerdp-codec1.1
  libfreerdp-common1.1.0 libfreerdp-core1.1 libfreerdp-crypto1.1
  libfreerdp-gdi1.1 libfreerdp-locale1.1 libfreerdp-primitives1.1
  libfreerdp-utils1.1 libgcc-6-dev libgom-1.0-common libllvm3.8 libobjc-6-dev
  libobjc4 libsfcgal1 libtinfo-dev libtxc-dxtn-s2tc libwinpr-crt0.1
  libwinpr-crypto0.1 libwinpr-dsparse0.1 libwinpr-environment0.1
  libwinpr-file0.1 libwinpr-handle0.1 libwinpr-heap0.1 libwinpr-input0.1
  libwinpr-interlocked0.1 libwinpr-library0.1 libwinpr-path0.1
  libwinpr-pool0.1 libwinpr-registry0.1 libwinpr-rpc0.1 libwinpr-sspi0.1
  libwinpr-synch0.1 libwinpr-sysinfo0.1 libwinpr-thread0.1 libwinpr-utils0.1
  linux-libc-dev llvm-3.8 llvm-3.8-dev llvm-3.8-runtime python-all python-bson
  python-bson-ext python-dbus python-gridfs python-keyring python-keyrings.alt
  python-pip python-pip-whl python-pymongo python-pymongo-ext
  python-secretstorage python-wheel
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  cpp-7 gcc-7 gcc-7-base gcc-8-base lib32gcc1 libasan4 libatomic1 libc-bin
  libc-dev-bin libc6 libc6-dbg libc6-i386 libcc1-0 libcilkrts5 libexpat1
  libgcc-7-dev libgcc1 libgfortran4 libgomp1 libgtksourceview2.0-0
  libgtksourceview2.0-common libitm1 liblsan0 libmpc3 libmpfr6 libmpx2
  libobjc4 libquadmath0 libstdc++6 libtsan0 libubsan0 locales
  python-gtksourceview2
Suggested packages:
  gcc-7-locales gcc-7-multilib gcc-7-doc libgcc1-dbg libgomp1-dbg libitm1-dbg
  libatomic1-dbg libasan4-dbg liblsan0-dbg libtsan0-dbg libubsan0-dbg
  libcilkrts5-dbg libmpx2-dbg libquadmath0-dbg glibc-doc
  libgtksourceview2.0-dev
Recommended packages:
  libc6-dev
**The following packages will be REMOVED:
  bluelog build-essential clang clang-3.8 g++ g++-7 glusterfs-common golismero
  hashcat kali-linux-full libacl1-dev libattr1-dev libbluetooth-dev libc6-dev
  libexpat1-dev libpocl1 libpocl1-common libpython-all-dev libpython-dev
  libpython2.7-dev libstdc++-6-dev libstdc++-7-dev pocl-opencl-icd
  python-all-dev python-dev python2.7-dev**
The following NEW packages will be installed:
  gcc-8-base lib32gcc1 lib32stdc++6 libc6-i386 libgtksourceview2.0-0
  libgtksourceview2.0-common libmpfr6 python-gtksourceview2
The following packages will be upgraded:
  cpp-7 gcc-7 gcc-7-base libasan4 libatomic1 libc-bin libc-dev-bin libc6
  libc6-dbg libcc1-0 libcilkrts5 libexpat1 libgcc-7-dev libgcc1 libgfortran4
  libgomp1 libitm1 liblsan0 libmpc3 libmpx2 libobjc4 libquadmath0 libstdc++6
  libtsan0 libubsan0 locales
26 upgraded, 8 newly installed, 26 to remove and 216 not upgraded.
Need to get 40.9 MB of archives.
After this operation, 298 MB disk space will be freed.
Do you want to continue? [Y/n]
--> My system is 64bit.
--> OS installed : $ uname -a
Linux mypc 4.13.0-kali1-amd64 #1 SMP Debian 4.13.10-1kali2 (2017-11-08) x86_64 GNU/Linux
(kali linux 2017.3)

**Comment by emdete** [comment]
it seems `android-arm-*/linux-x64` contains 32bit executables (as `emulator/lib64/libstdc++` from android sdk which i just deleted to use the system installed ones). i just did a

```
cd flutter/bin/cache/artifacts/engine
cp android-arm64-profile/linux-x64/gen_snapshot android-arm-profile/linux-x64/gen_snapshot
cp android-arm64-release/linux-x64/gen_snapshot android-arm-release/linux-x64/gen_snapshot
```

as a workaround. after that `flutter doctor` is fine and i can build & run the sample app on my android phone.

would be good to know why 32bit is on a 64bit system and why it is tried to be loaded.

**Comment by MrSlayers** [comment]
On Fedora 27 I tried

``
dnf install libstdc++.i686
``

But already installed on my system...

**Comment by MrSlayers** [comment]
With https://github.com/flutter/flutter/issues/6207#issuecomment-373100050 comment that's work !

**Comment by benbucksch** [comment]
Ubuntu 16.04 with all updates:
```
# apt-get update
# apt-get upgrade
# apt-get install lib32stdc++6
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 lib32stdc++6 : Depends: gcc-8-base (= 8-20180414-1ubuntu2) but 8-20180424-0ubuntu1~16.04.1 is to be installed
                Depends: lib32gcc1 (>= 1:4.2) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by funkybaboon** [comment]
How do I fix this on centos7? Installing `libstdc++` ist not working.

```
[root@9f4595cf5fe7 /]# yum -y -q install libstdc++
Package libstdc++-4.8.5-28.el7_5.1.x86_64 already installed and latest version
[root@9f4595cf5fe7 /]# flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel unknown, v0.5.7, on Linux, locale en_US)
    ✗ Downloaded executables cannot execute on host.
      See https://github.com/flutter/flutter/issues/6207 for more information
      On Debian/Ubuntu/Mint: sudo apt-get install lib32stdc++6
      On Fedora: dnf install libstdc++.i686
      On Arch: pacman -S lib32-libstdc++5
      
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✗] Android Studio (not installed)
[!] Connected devices
    ! No devices available

! Doctor found issues in 3 categories.
```

**Comment by funkybaboon** [comment]
With this workaround posted earlier I can build on centos7: https://github.com/flutter/flutter/issues/6207#issuecomment-373100050

**Comment by JakeRed** [comment]
After reading the above answer, I couldn't find a way for MacOS to successfully run the flutter doctor command. Has anyone succeeded? this is error info：

[!] Flutter (Channel beta, v0.6.0, on Mac OS X 10.13.6 17G65, locale zh-Hans-CN)
    • Flutter version 0.6.0 at /Users/yimutian/flutter
    • Framework revision 9299c02cf7 (3 weeks ago), 2018-08-16 00:35:12 +0200
    • Engine revision e3687f70c7
    • Dart version 2.1.0-dev.0.0.flutter-be6309690f
    ✗ Downloaded executables cannot execute on host.
      See https://github.com/flutter/flutter/issues/6207 for more information
      

[!] Android toolchain - develop for Android devices (Android SDK 28.0.0)
    • Android SDK at /Users/yimutian/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

Looking forward to your help！

**Comment by cbracken** [comment]
@JakeRed your error looks unrelated to this bug, which is Linux-specific. Apologies for the confusing error message. It seems as though your Flutter install is somehow corrupt. Can you try the following from a terminal window and see if it helps?
```
rm -rf ~/flutter/bin/cache
flutter doctor --verbose
```

Based on your locale, in case you're in China, it may be useful to follow our instructions for [using Flutter in China](https://github.com/flutter/flutter/wiki/Using-Flutter-in-China).

**Comment by JakeRed** [comment]
@cbracken  Thank you for your answer, Your answer has solved my problem.
I am in China, but I use VPN.

**Comment by oswee** [comment]
`sudo dnf install libstdc++.i686`
....
`Error: Transaction check error:
  file /usr/share/gcc-8/python/libstdcxx/v6/__pycache__/printers.cpython-37.opt-1.pyc from install of libstdc++-8.2.1-5.fc29.i686 conflicts with file from package libstdc++-8.2.1-4.fc29.x86_64
  file /usr/share/gcc-8/python/libstdcxx/v6/__pycache__/printers.cpython-37.pyc from install of libstdc++-8.2.1-5.fc29.i686 conflicts with file from package libstdc++-8.2.1-4.fc29.x86_64
  file /usr/share/gcc-8/python/libstdcxx/v6/printers.py from install of libstdc++-8.2.1-5.fc29.i686 conflicts with file from package libstdc++-8.2.1-4.fc29.x86_64`
I have some Apache Kafka Client, Docker, Go, stuff installed and i need it.

**Comment by metaa** [comment]
I'm on Windows 10 1903 and this still seems to be an issue.

To resolve this partially, I installed the Dart SDK with [Chocolatey](https://chocolatey.org/packages/dart-sdk) and replaced the Dart SDK in the Flutter SDK directory with a junction to the Chocolatey Dart SDK directory (happens to be `C:\tools\dart-sdk` in my case).

I guess a manual install of the Dart SDK would work too, but this was a simple solution in my opinion.

Even though `pub upgrade` now works, it fails to build "the flutter" tool still.

**Comment by MaterialTime** [comment]
I'm facing the same issue as janispritzkau

**Comment by lafriakh** [comment]
@janispritzkau @MaterialTime I solve the same problem from Solus documentation.

Just install the Prerequisites in the documentation.

**Comment by MaterialTime** [comment]
@lafriakh What documentation? Could you provide a link?

**Comment by lafriakh** [comment]
> @lafriakh What documentation? Could you provide a link?

You can find it here: https://getsol.us/articles/software/building-android/en/

**Comment by MaterialTime** [comment]
> > @lafriakh What documentation? Could you provide a link?
> 
> You can find it here: https://getsol.us/articles/software/building-android/en/

Thanks, that's fixed the issue.

**Comment by qiyaozu** [comment]
> After reading the above answer, I couldn't find a way for MacOS to successfully run the flutter doctor command. Has anyone succeeded? this is error info：
> 
> ```
> [!] Flutter (Channel beta, v0.6.0, on Mac OS X 10.13.6 17G65, locale zh-Hans-CN)
>     • Flutter version 0.6.0 at /Users/yimutian/flutter
>     • Framework revision 9299c02cf7 (3 weeks ago), 2018-08-16 00:35:12 +0200
>     • Engine revision e3687f70c7
>     • Dart version 2.1.0-dev.0.0.flutter-be6309690f
>     ✗ Downloaded executables cannot execute on host.
>       See https://github.com/flutter/flutter/issues/6207 for more information
>       
> 
> [!] Android toolchain - develop for Android devices (Android SDK 28.0.0)
>     • Android SDK at /Users/yimutian/Library/Android/sdk
>     • Android NDK location not configured (optional; useful for native profiling support)
>     • Platform android-28, build-tools 28.0.0
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
>     ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
> ```
> 
> Looking forward to your help！

I have the same problem,Have you solved it yet?

**Comment by BradHoffman96** [comment]
I am experiencing this issue on the new macOS 10.15 Beta. Just thought I would give a heads up.

**Comment by xascis** [comment]
Getting the same warning on macOS 10.15.

I can run the app in Android and iOS emulator, build ios app but can't build apk. This is the log I get after running `flutter build apk`:

```
Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter build aot --suppress-analytics --quiet --target lib/main.dart --output-dir /Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm --target-platform android-arm --release

## exception

ProcessException: ProcessException: Bad CPU type in executable
  Command: /Users/francisco/Developer/flutter/bin/cache/artifacts/engine/android-arm-release/darwin-x64/gen_snapshot --causal_async_stacks --deterministic --snapshot_kind=app-aot-blobs --vm_snapshot_data=/Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm/vm_snapshot_data --isolate_snapshot_data=/Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm/isolate_snapshot_data --vm_snapshot_instructions=/Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm/vm_snapshot_instr --isolate_snapshot_instructions=/Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm/isolate_snapshot_instr --no-sim-use-hardfp --no-use-integer-division /Users/francisco/Developer/marvel_comics/build/app/intermediates/flutter/release/android-arm/app.dill

```
```
#0      runCommandAndStreamOutput (package:flutter_tools/src/base/process.dart:142:27)
<asynchronous suspension>
#1      GenSnapshot.run (package:flutter_tools/src/base/build.dart:65:12)
#2      AOTSnapshotter.build.<anonymous closure> (package:flutter_tools/src/base/build.dart:197:97)
#3      AOTSnapshotter._timedStep (package:flutter_tools/src/base/build.dart:328:33)
<asynchronous suspension>
#4      AOTSnapshotter.build (package:flutter_tools/src/base/build.dart:197:43)
<asynchronous suspension>
#5      BuildAotCommand.runCommand (package:flutter_tools/src/commands/build_aot.dart:149:56)
<asynchronous suspension>
#6      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:488:18)
<asynchronous suspension>
#7      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:424:33)
<asynchronous suspension>
#8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#9      _rootRun (dart:async/zone.dart:1124:13)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _runZoned (dart:async/zone.dart:1516:10)
#12     runZoned (dart:async/zone.dart:1463:12)
#13     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#14     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:415:20)
#15     CommandRunner.runCommand (package:args/command_runner.dart:196:27)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:396:21)
<asynchronous suspension>
#17     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#18     _rootRun (dart:async/zone.dart:1124:13)
#19     _CustomZone.run (dart:async/zone.dart:1021:19)
#20     _runZoned (dart:async/zone.dart:1516:10)
#21     runZoned (dart:async/zone.dart:1463:12)
#22     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#23     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:356:19)
<asynchronous suspension>
#24     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:111:29)
#25     new Future.sync (dart:async/future.dart:224:31)
#26     CommandRunner.run (package:args/command_runner.dart:111:11)
#27     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:242:18)
#28     run.<anonymous closure> (package:flutter_tools/runner.dart:60:20)
<asynchronous suspension>
#29     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#30     _rootRun (dart:async/zone.dart:1124:13)
#31     _CustomZone.run (dart:async/zone.dart:1021:19)
#32     _runZoned (dart:async/zone.dart:1516:10)
#33     runZoned (dart:async/zone.dart:1463:12)
#34     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#35     runInContext (package:flutter_tools/src/context_runner.dart:54:24)
<asynchronous suspension>
#36     run (package:flutter_tools/runner.dart:51:10)
#37     main (package:flutter_tools/executable.dart:62:9)
<asynchronous suspension>
#38     main (file:///Users/francisco/Developer/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#39     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:298:32)
#40     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:171:12)
```

## flutter doctor

```
[!] Flutter (Channel dev, v1.7.0, on Mac OS X 10.15 19A471t, locale es-ES)
    • Flutter version 1.7.0 at /Users/francisco/Developer/flutter
    • Framework revision f36a35d20a (4 days ago), 2019-05-31 15:27:56 -0400
    • Engine revision a32df2c928
    • Dart version 2.3.2 (build 2.3.2-dev.0.0 445a23a9bc)
    ✗ Downloaded executables cannot execute on host.
      See https://github.com/flutter/flutter/issues/6207 for more information


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/francisco/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11M336w
    • CocoaPods version 1.7.1

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[!] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.34.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.1.0

[!] Connected device
    ! No devices available

! Doctor found issues in 3 categories.
```

**Comment by kangwang1988** [comment]
If someone is visiting this issue when `flutter doctor` fails on a MacOS 10.15, please refer to:
https://github.com/flutter/flutter/issues/22598

**Comment by sunbinnnnn** [comment]
For opensuse, use `sudo zypper install libstdc++6-32bit` to deal.

**Comment by AndreaMiotto** [comment]
any update with this issue on mac os 10.15?

**Comment by benbucksch** [comment]
@AndreaMiotto : See 2 comments above yours. Mac 10.15 is #22598.

**Comment by herawais** [comment]
The following packages have unmet dependencies:
lib32stdc++6 : Depends: gcc-9-base (= 9-20190402-1ubuntu1) but 9.1.0-2ubuntu2~19.04 is to be installed
Depends: lib32gcc1 (>= 1:4.2) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.


on Ubuntu 19.04
